proposals/drafts/ocdid_governance.rst

```diff
@@ -0,0 +1,113 @@
+
+
+Implementation
+==============
+
+Roles & Responsibilities
+------------------------
+``User``
+~~~~~~~~
+Anyone using OCDIDs may contribute through Slack/Google group discussion, reporting issues, or assisting with subject-matter expertise.
+These channels for communication will be publicaly displayed on the Open Civic Data website.
+
+``Contributor``
+~~~~~~~~~~~~~~~
+Any individual or organization ``user`` may become a ``contributor`` by submitting a pull request adding, correcting, or aliasing jurisdictional OCDIDs.
+First time ``contributors`` will be advised to consult with an existing ``committer`` before submitting their first PR.
+
+``Committer`` 
+~~~~~~~~~~~~~
+``Committers`` have the ability to approve and commit pull requests from ``contributors`` and other ``committers``. 
+An initial cohort of ``committers`` will be determined by current project contributors.
+Any individual or organization who is a ``contributor`` and who agrees to the repsonsibilities listed here may apply to become a ``committer``. 
+
+* There will be a two month period where existing ``committers`` may raise objections to an applicant, otherwise the applying organization/individual will be considered approved.
+* In the case that an objection is raised, a discussion and vote will be had amongst existing ``committers``, with a simple majority necessary for approval.
+
+``Committers`` will be approved on an organizational basis, unless explicitly noted for an individual.
+
+* Multiple members of an organization may have committing ability, but the organization itself will be counted for all voting and approval processes.
+* Organizations will be asked to designate a primary point of contact who will act on behalf of the organization in procedural matters.
+
+``Committers`` are expected to participate in approval, support, maintenance, and other community related activities or may have their ``committer`` status revoked. 
+
+* ``Committers`` will have their status reviewed on a calendar year basis. 
+
+
+Contribution Process
+--------------------
+General contributions
+~~~~~~~~~~~~~~~~~~~~~
+Any ``contributor`` may create a pull request for generative IDs, corrections, aliases, etc. 
+
+* A pull request from a ``contributor`` will be considered accepted when reviewed by and approved by two ``committers``.
+* If the contribution is from a current ``committer`` in good standing, only one additional ``committer`` review is necessary.
+* No two members of the same organization may be involved in the acceptance of a pull request.
+
+Commits should be reviewed within 2 weeks of a pull request. Accellerated timeline needs should be communicated via Slack/email.
+
+* If, after two months there is neither a formal approval or an ongoing conversation around a request, that request will be considered accepted.
```

Comment:
Even if the person is not a committer, but just a contributor?

Reply:
This seems dangerous. I understand that timely review is a present issue, but once we implement the rest of the steps, it should be largely mitigated. (Introducing this raises the worst-case scenario of being bound to accept a PR that deletes all files.)

Reply:
@fgregg I'm thinking yes, as long as they're not a new contributor (which shouldn't be an issue, assuming the new committer is following the 'consult with a current committer' guideline in the COMMITTING.rst). I'm imagining the scenario where this most often would go into effect for a contributor would be for minor changes/corrections, as folks who would be making larger changes would probably either want to become or will be part of an organization that at least has a committer. If you think that's a bad assumption, or have strong feelings otherwise, happy to continue discussing.

Reply:
@jpmckinney I hear you and agree that it should be largely mitigated, but still think having an explicit backstop against things languishing in PR-purgatory is important. What if instead of straight auto-approve, a PR that hasn't been touched in 2 months gets pro-actively sent to the whole committer group for a "speak now or forever hold your peace" style review? It can still be approved by silence, if, say, it's about an esoteric thing that no current committers have knowledge of/an opinion on, but it ensures there are eyes on it to avoid a "delete everything" scenario?

Reply:
I'm not familiar with explicit backstops in open source development. Is there precedent for that? (preferably larger projects)

If literally no committer has an interest or opinion, I think that's to be solved by adding more knowledgeable committers – not by silently approving PRs. Divisions are political. Let's take fictional Kharun and South Kharun from the show Borgen. We get a PR that puts a division in the hierarchy of one country. But in fact, it's contested territory. I'd rather have someone with  political acuity chime in, than approve that change just because no present committer cares.

In the real world, if we got a PR for `ocd-division/country:cn/province:tw` for example (making Taiwan a province of China), it would at least make me think before merging. Whether we'd issue both that and `ocd-division/country:tw` to satisfy both views on the matter, I don't know, but I know it'd require some care, and shouldn't be automatic.

Reply:
Those are both interesting cases that I agree would be tricky to handle, though even in the construction I mentioned above could still be prevented from getting automatically accepted by someone saying "I'm not sure what's going on here but maybe we should have a longer conversation/reach out to an expert." 

What I see as a more common way this would arise, though admittedly it may only happen in the US, is this: a researcher's gets on the phone with Elko, Nevada and discovers they have a separately elected [television control board](http://www.elkocountynv.net/departments/clerk/2012%20TV%20District%208%20of%209.pdf) that not only has it's own distinct geography, but it has [geographic subdistricts](https://www.dropbox.com/s/ow4c4u601a9n4gd/Screenshot%202017-11-01%2019.15.04.JPG?dl=0) from which people are elected. They create``ocd-division/country:us/state:nv/county:elko/television:e`` (or some such) and want to add it to the repo. Repeat any number of times for any number of weird districts, precinct name changes, etc. I completely get why someone wouldn't feel comfortable giving that an affirmative approval on face, but at some point it starts affecting how comprehensive tools using OCDIDs can be.

If there's a better way of handling that sort of use case that I'm missing, I'm all for it. Otherwise I think the sheer volume of those cases, in the US at least,  means the benefit of allowing them a graceful way to get in outweighs the potential for harm.

Reply:
I think this is partly solved by having (geographically scoped) committers who are more lax with respect to performing verification and who are happy to merge things from people who appear to have good knowledge of a subject area. Basically, I still think this problem is solved at the level of people (who is deciding to merge), not at the level of process (when to auto-merge, etc.).

As we progress further down the hierarchy, the consequences of a bad merge become more and more limited. So, I'm fine with the governance of television control boards divisions being, "You say it exists? Sure! I'll merge."

We have some identifiers for countries for which no committer has sophisticated knowledge, and we have no identifiers for most countries. I think, for all such countries, we need an exemption from any sort of process-based auto-merge-like rule, and we also need an exemption for the top-level of the OCDID hierarchy (`ocd-division/country:`). Failure to adequately protect those with people-based governance can severely damage the project.

I'd also be in favor of scoping the auto-merge-like rule to specific geographies and/or specific division types (like `television`). I don't think the type `state` in the US should fall under the auto-merge-like rule – but I'll let US committers decide. For Canada, there's very low demand for PRs, so I'm sure I need to implement such a rule at this time.